language/compiler/ir_to_bytecode/syntax/src/ast.rs

```diff
@@ -523,6 +526,23 @@ impl Script {
             FunctionBody::Native => panic!("main() can't be native"),
         }
     }
+
+    /// Return a vector of `AccessPath` where the on chain dependent modules are stored.
+    pub fn get_on_chain_deps(&self) -> Vec<AccessPath> {
```

Comment:
Would it make more sense for this to return a vector of `ModuleId`'s? That way, we could implement the same function for `CompiledModule` and share the `ModuleId` -> `AccessPath` logic in both.

Also: perhaps name something like `get_module_dependencies`? The `on_chain` part is a bit confusing to me given that this doesn't actually read from the blockchain...